webapp/src/ts/services/feedback.service.ts

```diff
@@ -43,6 +62,56 @@ export class FeedbackService {
       .filter(i => !!i);
   }
 
+  private shouldGenerateFeedback(message:string, exceptionMessage:string, exception?) {
+    // requiring a valid error to be logged to avoid cascades of feedback docs
+    if (!message || !exception) {
+      return false;
+    }
+
+    // don't double-log errors as a basic infinite loop defense
+    if (this.lastErrorMessage === message || this.lastErrorMessage === exceptionMessage) {
+      return false;
+    }
+
+    const matchesNoFeedback = this.NO_FEEDBACK_MESSAGES
+      .find((item:RegExp) => item.test(message) || item.test(exceptionMessage));
+
+    return !matchesNoFeedback;
+  }
+
+  private getExceptionMessage(exception) {
+    if (!exception) {
+      return;
+    }
+
+    if (typeof exception.error === 'string') {
+      return exception.error;
+    }
+
+    return exception.reason || exception.message;
+  }
+
+  private async generateFeedbackOnError(level, ...args) {
+    if (this.FEEDBACK_LEVEL !== level) {
+      return false;
+    }
+
+    const exception = args.find(arg => arg instanceof Error || arg?.stack || arg instanceof HttpErrorResponse);
+    const exceptionMessage = this.getExceptionMessage(exception);
+    const loggedMessage = String(args[0]);
+
+    if (this.shouldGenerateFeedback(loggedMessage, exceptionMessage, exception)) {
+      const message = exceptionMessage || loggedMessage;
+      this.lastErrorMessage = message;
+      try {
+        await this.createAndSave( { message, stack: exception?.stack });
```

Comment:
If a loggedMessage and exceptionMessage are both available it would be ideal to capture both in the feedback doc. Is this possible?

You could take this even further by capturing all args (or all args that aren't objects?) so it could capture all the given detail, for example handling a call like this:

```
console.error('Location failed', `Error code ${code}`, err);
```

Reply:
Yea, I thought about this as well. I'm sort of worried of breaking any existent queries that work with feedback docs. Do you think I should just add an additional field with the message?

Reply:
Done.